DB/src/main/java/io/deephaven/db/v2/sources/chunkcolumnsource/CharChunkColumnSource.java

```diff
@@ -101,19 +108,35 @@ public void fillPrevChunk(@NotNull FillContext context, @NotNull WritableChunk<?
         fillChunk(context, destination, orderedKeys);
     }
 
+    /**
+     * Given an index within this column's address space; return the chunk that contains the index.
+     *
+     * @param start the data index to find the corresponding chunk for
+     *
+     * @return the chunk index within data and offsets
+     */
     private int getChunkIndex(final long start) {
         return getChunkIndex(start, 0);
     }
 
-    private int getChunkIndex(final long start, final int startIndex) {
-        int index = offsets.binarySearch(start, startIndex, offsets.size());
+    /**
+     * Given an index within this column's address space; return the chunk that contains the index.
+     *
+     * @param start the data index to find the corresponding chunk for
+     * @param startChunk the first chunk that may possibly contain start
+     *
+     * @return the chunk index within data and offsets
+     */
+
+    private int getChunkIndex(final long start, final int startChunk) {
+        int index = offsets.binarySearch(start, startChunk, offsets.size());
         if (index < 0) {
             index = -index - 2;
         }
         return index;
     }
 
-    public void addChunk(final CharChunk<? extends Attributes.Values> chunk) {
+    private void addChunk(final CharChunk<? extends Attributes.Values> chunk) {
```

Comment:
Why make this private? Wouldn't we rather code have the option to use the typed version?

Reply:
It wasn't used outside, so by making it private I didn't need to duplicate the Javadoc.  I could consider making it public again if it would make sense from the producer.

Reply:
We can punt until we have a use-case.